app/src/main/java/ca/nick/multimodulenavigation/MainActivity.kt

```diff
@@ -17,8 +17,15 @@ class MainActivity : AppCompatActivity() {
 
         if (supportFragmentManager.findFragmentByTag("list") == null) {
             supportFragmentManager.beginTransaction()
-                .replace(R.id.container, ListFragment::class.java, null, "list")
+                .replace(
+                    R.id.container,
+                    ListFragment(
+                        onItemClicked = OnItemClickedDirections(supportFragmentManager),
+                        SomeHeavyWeightDependency()
+                    ),
+                    "list"
+                )
```

Comment:
instantiating a ListFragment direction will crash on process recreation, it has to go thru the FragmentFactory. this new API can be used instead.

```suggestion
                .replace<ListFragment>(R.id.container, "list")
```